src/Modal.js

```diff
@@ -0,0 +1,170 @@
+// @flow
+
+import React from 'react';
+import classNames from 'classnames';
+import './Modal.scss';
+
+type ModalProps = {
+    show: boolean,
+    focusElementSelector: string,
+    focusOnCloseSelector: string,
+    ariaLabel?: string,
+    ariaLabelledById?: string,
+    ariaDescribedById?: string,
+    children?: any,
+    onClose: () => void
+};
+
+class Modal extends React.Component<ModalProps, {}> {
+    modalRef: { current: null | Element };
+    lastFocus: any;
+    ignoreFocusChanges: boolean;
+    constructor(props: ModalProps) {
+        super(props);
+        this.modalRef = React.createRef();
+        this.lastFocus = document.querySelector(this.props.focusElementSelector);
+        this.ignoreFocusChanges = false;
+    }
+
+    focusFirstDescendant = (element: any) => {
+        for (let i = 0; i < element.childNodes.length; i++) {
+            const child = element.childNodes[i];
+            if (this.attemptFocus(child) ||
+                this.focusFirstDescendant(child)
+            ) {
+                return true;
+            }
+        }
+        return false;
+    };
+
+    focusLastDescendant = (element: any) => {
+        for (let i = element.childNodes.length - 1; i >= 0; i--) {
+            const child = element.childNodes[i];
+            if (this.attemptFocus(child) ||
+                this.focusLastDescendant(child)
+            ) {
+                return true;
+            }
+        }
+        return false;
+    };
+
+    attemptFocus = (element: HTMLElement): boolean => {
+        this.ignoreFocusChanges = true;
+        // $FlowFixMe: properties type and checked is missing in HTMLElement
+        if (element.type === 'radio' && !element.checked) {
+            return false;
+        }
+        if (element.focus) {
+            element.focus();
+        }
+        this.ignoreFocusChanges = false;
+        return (document.activeElement === element);
+    };
+
+    handleFocusTrap = (event: Event) => {
+        if (this.ignoreFocusChanges) {
+            return;
+        }
+        // Add guard cases to check if properties exist
```

Comment:
What does this comment mean?

Reply:
I think I left a note for myself and this comment seems meaningless now. I removed the comment.